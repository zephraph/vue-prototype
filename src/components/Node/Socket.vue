<template lang="pug">
g.Socket(:transform="translate")
  circle(r="5")
  text(
    :x="output ? -8 : 8" y="4",
    :text-anchor="output ? 'end' : 'start'"
  ) {{ label }}
</template>

<script>
import translatable from 'mixins/translatable';

export default {
  mixins: [translatable],
  data() {
    return {
      y: this.offset || 40
    }
  },
  computed: {
    x() {
      return this.output
        ? this.width
        : 0
    },
    output() {
      return this.type === 'output';
    }
  },
  props: [
    'label',
    'type', // input | output
    'width',
    'offset'
  ]
}
</script>

<style lang="stylus" scoped>
</style>
