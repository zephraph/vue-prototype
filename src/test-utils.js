import { createRenderer } from 'vue-server-renderer';
import Vue from 'vue';

export const render = async renderFunc => {
  const renderer = createRenderer();
  return new Promise((resolve, reject) => {
    renderer.renderToString(new Vue({ render: renderFunc }), (err, str) => {
      if (err) reject(err);
      resolve(str);
    });
  });
}
