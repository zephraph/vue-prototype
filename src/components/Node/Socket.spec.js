import { render } from 'test-utils';
import Socket from './Socket.vue';

test('Socket component should render in correct place for input', async () => {
  const result = await render(h => <Socket width={ 10 }/>);
  expect(result).toMatchSnapshot();
});

test('Socket component should render in correct place for output', async () => {
  const result = await render(h => <Socket type="output" width={ 10 }/>);
  expect(result).toMatchSnapshot();
});
