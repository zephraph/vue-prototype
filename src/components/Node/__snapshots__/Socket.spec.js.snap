exports[`test Socket component should render in correct place for input 1`] = `
<g transform="translate(0,40)" server-rendered="true" class="Socket">
    <circle r="5"></circle><text x="8" y="4" text-anchor="start"></text></g>
`;

exports[`test Socket component should render in correct place for output 1`] = `
<g transform="translate(10,40)" server-rendered="true" class="Socket">
    <circle r="5"></circle><text x="-8" y="4" text-anchor="end"></text></g>
`;
